src/main/kotlin/app/revanced/patches/youtube/misc/parseuriredirect/patch/ParseUriRedirectPatch.kt

```diff
@@ -0,0 +1,72 @@
+package app.revanced.patches.youtube.misc.parseuriredirect.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patches.youtube.misc.integrations.patch.IntegrationsPatch
+import app.revanced.patches.youtube.misc.parseuriredirect.annotations.ParseUriRedirectCompatibility
+import app.revanced.patches.youtube.misc.parseuriredirect.fingerprints.ParseUriRedirectFirstFingerprint
+import app.revanced.patches.youtube.misc.parseuriredirect.fingerprints.ParseUriRedirectSecondFingerprint
+import app.revanced.patches.youtube.misc.settings.bytecode.patch.SettingsPatch
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.StringResource
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.SwitchPreference
+import org.jf.dexlib2.iface.instruction.formats.Instruction35c
+import org.jf.dexlib2.iface.instruction.formats.Instruction11x
+
+@Patch
+@DependsOn([IntegrationsPatch::class])
+@Name("uri-redirect")
+@Description("Follow direct links, bypassing youtube.com/redirect.")
+@ParseUriRedirectCompatibility
+@Version("0.0.1")
+class ParseUriRedirectPatch : BytecodePatch(
```

Comment:
Make sure to apply the changes everywhere else:
```suggestion
class OpenLinksDirectlyPatch : BytecodePatch(
```